crypto/evp/keymgmt_meth.c

```diff
@@ -54,11 +54,31 @@ static void *keymgmt_from_dispatch(int name_id,
             keymgmt->importdomparams =
                 OSSL_get_OP_keymgmt_importdomparams(fns);
             break;
+        case OSSL_FUNC_KEYMGMT_GENDOMPARAMS_INIT:
```

Comment:
Perhaps rewrite this set of cases as
```
if (keymgmt->foo == NULL)
   keymgmt->foo = OSSL_get_OP_keymgmt_foo(fns);
break;
```
?

Reply:
Maybe you should look at the whole function first? In this case, line 43...

Reply:
Yes, I did misunderstand. Sorry.

Good suggestion.